File: medium/python/462.py
# https://leetcode.com/problems/minimum-moves-to-equal-array-elements-ii/description/

class Solution:
    def minMoves2(self, nums: List[int]) -> int:
        nums.sort()

        l, r = 0, len(nums) - 1
        res = 0

        while l < r:
            res += nums[r] - nums[l]
            l += 1
            r -= 1

        return res
        

File: medium/cpp/462.cpp
// https://leetcode.com/problems/minimum-moves-to-equal-array-elements-ii/description/

class Solution
{
public:
    int minMoves2(std::vector<int>& nums)
    {
        std::sort(std::begin(nums), std::end(nums));

        const auto n = std::size(nums);

        auto pref = 0LL;
        auto suff = std::accumulate(std::cbegin(nums), std::cend(nums), 0LL);

        auto res = -1LL;

        for (auto i = 0; i < n; i++)
        {
            suff -= nums[i];

            const auto val = 1LL * i * nums[i] - pref + suff - 1LL * (n - i - 1) * nums[i];
            res = res == -1 ? val : std::min<long long>(val, res);

            pref += nums[i];
        }

        return res;
    }
};